Base/Python/slicer/parameterNodeWrapper/parameterPack.py

```diff
@@ -0,0 +1,285 @@
+        membertype, annotations = splitAnnotations(nametype)
+
+        serializer, annotations = createSerializer(membertype, annotations)
+        default, annotations = extractDefault(annotations)
+        default = default.value if default is not None else serializer.default()
+
+        if annotations:
+            print("Warning: unused annotations", annotations)
+        try:
+            serializer.validate(default)
+        except Exception as e:
+            raise ValueError(f"The default parameter of '{default}' fails the validation checks:\n  {str(e)}")
+
+        parameter = _ParameterInfo(name, serializer, default)
+        # note: heavily relying on the fact that dict is ordered
+        allParameters[name] = parameter
+
+        for reserved in [_implName(parameter.basename), _serializerName(parameter.basename)]:
+            if reserved in classtype.__dict__ or reserved in members.keys():
+                raise ValueError(f"Cannot use reserved name (based off of parameter name) '{reserved}' in a parameterPack")
+
+        setattr(classtype, parameter.basename, _makeConcreteProperty(parameter.basename))
+        setattr(classtype, _serializerName(parameter.basename), serializer)
+    
+    # give default methods if not already specified
+    if "__init__" not in classtype.__dict__:
+        setattr(classtype, "__init__", _initMethod)
+    if "__eq__" not in classtype.__dict__:
+        setattr(classtype, "__eq__", _eqMethod)
+    if "__str__" not in classtype.__dict__:
+        setattr(classtype, "__str__", _strMethod)
+    if "__repr__" not in classtype.__dict__:
+        setattr(classtype, "__repr__", _strMethod)
+
+    # required to make this class work, these names are reserved
+    for reserved in ["_parameterPack_allParameters", "_is_parameterPack"]:
+        if reserved in classtype.__dict__:
+            raise ValueError(f"Cannot use reserved name '{reserved}' in a parameterPack")
+    setattr(classtype, "_parameterPack_allParameters", allParameters)
+    setattr(classtype, "_is_parameterPack", True)
+    return classtype
+
+
+def parameterPack(classtype=None):
+    """
+    Class decorator to make an parameterPack.
+    """
+    def wrap(cls):
+        return _processParameterPack(cls)
+
+    # See if we're being called as @parameterPack or @parameterPack().
+    if classtype is None:
+        return wrap
+    return wrap(classtype)
+
+
+class ObservedParameterPack:
+    """
+    Class with the ability to observe any parameterPack and write on change.
+    """
+    def __init__(self, parameterNode, serializer: Serializer, name: str, startingValue):
+        # Because we are overriding __getattr__ and __setattr__ to pass through to _value
+        # we need to access this class's members directly through __dict__
+        self.__dict__['_parameterNode'] = parameterNode
+        self.__dict__['_serializer'] = serializer
+        self.__dict__['_name'] = name
+        self.__dict__['_value'] = startingValue
+
```

Comment:
_Generally_ I think it's a smell to manually mess with `__dict__`.  I'd prefer to use `super().__setattr__` and `super().__getattr__`; that'd also set you implement `__getattribute__` instead.

Reply:
Updated to use super.